src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [state];
+
+  for (let i = 0; i < actions.length; i++) {
+    const prevState = result[i];
+    let newState;
+
+    switch (actions[i].type) {
+      case 'addProperties':
+        newState = Object.assign({}, prevState, actions[i].extraData);
```

Comment:
here you can use 
`currentState = { ...currentState, ...actions[i].extraData }`
please check how you can optimize your code with this hints